clang/lib/Basic/Targets/X86.h

```diff
@@ -719,13 +719,33 @@ class LLVM_LIBRARY_VISIBILITY X86_64TargetInfo : public X86TargetInfo {
     Int64Type = IsX32 ? SignedLongLong : SignedLong;
     RegParmMax = 6;
 
+    bool isBIDEncoding = Opts.DFPEncoding == TargetOptions::BinaryEncoding::BID;
+    bool isDPDEncoding = Opts.DFPEncoding == TargetOptions::BinaryEncoding::DPD;
```

Comment:
At present, this information should already be available on the `TargetInfo` base class via the `DecimalFloatEnablementAndMode` member at this point (if `hasDecimalFloatingPoint()` returns true, then `getDecimalFloatingPointMode()` will return the active DFP mode).

Reply:
See comment above.